test/datastream/datastream_test.go

```diff
@@ -88,8 +93,10 @@ var _ = Describe("Streaming Data Conversion", func() {
 				sampleFilename, err := f.FormatTestData(fn, ff...)
 				Expect(err).NotTo(HaveOccurred(), "Error formatting test data.")
 
-				By(fmt.Sprintf("Confirming sample file name %q matches expected file name %q", sampleFilename, fn+strings.Join(ff, "")))
-				Expect(sampleFilename).To(Equal(fn+strings.Join(ff, "")), "Test data filename doesn't match expected file name.")
+				//By(fmt.Sprintf("Confirming sample file name %q matches expected file name %q", sampleFilename, fn+strings.Join(ff, "")))
+				By(fmt.Sprintf("Confirming sample file name %q matches expected file name %q base = %q", sampleFilename, fnbase+strings.Join(ff, ""), fnbase))
+				// Expect(sampleFilename).To(Equal(fn+strings.Join(ff, "")), "Test data filename doesn't match expected file name.")
+				Expect(sampleFilename).To(Equal(fnbase+strings.Join(ff, "")), "Test data filename doesn't match expected file name.")
```

Comment:
I think it's simpler and easier to maintain if you add the `ExpectedFName: string` to the test type, Eg:
```
...
{
	testDesc:      "should unpack .tar.xz",
	inFileName:    infilePath,
        expectFName: infilePath + ".tar.xz",
	expectFormats: []string{image.ExtTar, image.ExtXz},
},
{
	testDesc:      "should convert .qcow2",
	inFileName:    infilePath,
        expectFName: infilePath,
	expectFormats: []string{image.ExtQcow2},
},

```